File: Programming Fundamentals/C#/03.Arrays/Exercise/P07_MaxSequenceOfEqualElements/P07_MaxSequenceOfEqualElements.cs
namespace P07_MaxSequenceOfEqualElements
{
    using System;
    using System.Linq;

    class P07_MaxSequenceOfEqualElements
    {
        static void Main(string[] args)
        {
            int[] nums = Console.ReadLine()
                                .Split()
                                .Select(int.Parse)
                                .ToArray();

            int biggestNum = nums.Last();
            int biggestCount = 1;

            int currentNum = nums.Last();
            int currentCount = 1;

            for (int i = nums.Length - 1; i > 0; i--)
            {
                if (nums[i] != nums[i - 1])
                {
                    currentNum = nums[i - 1];
                    currentCount = 1;
                    continue;
                }

                currentCount++;

                if (currentCount >= biggestCount)
                {
                    biggestCount = currentCount;
                    biggestNum = currentNum;
                }
            }

            string sequence = string.Empty;
            for (int i = 0; i < biggestCount; i++)
            {
                sequence += $"{biggestNum} ";
            }

            Console.WriteLine(sequence.TrimEnd());
        }
    }
}

File: Programming Fundamentals/JS/03.Arrays/Exercise/p07_MaxSequenceOfEqualElements.js
function solve(nums = []) {
  let biggestNum = nums[nums.length - 1];
  let biggestCount = 1;

  let currentNum = nums[nums.length - 1];
  let currentCount = 1;

  for (let i = nums.length - 1; i > 0; i--) {
    if (nums[i] !== nums[i - 1]) {
      currentNum = nums[i - 1];
      currentCount = 1;
      continue;
    }

    currentCount++;

    if (currentCount >= biggestCount) {
      biggestCount = currentCount;
      biggestNum = currentNum;
    }
  }

  let sequence = '';
  for (let i = 0; i < biggestCount; i++) {
    sequence += `${biggestNum} `;
  }

  console.log(sequence.trimEnd());
}

// Don't copy the calling of the function in judge, you won't get any points, just the code above
solve([2, 1, 1, 2, 3, 3, 2, 2, 2, 1]);
solve([1, 1, 1, 2, 3, 1, 3, 3]);
solve([4, 4, 4, 4]);
solve([0, 1, 1, 5, 2, 2, 6, 3, 3]);
